samples/copy-from-container/pom.xml

```
@@ -0,0 +1,89 @@
+<project xmlns = "http://maven.apache.org/POM/4.0.0"
+   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+   xsi:schemaLocation = "http://maven.apache.org/POM/4.0.0
+   http://maven.apache.org/xsd/maven-4.0.0.xsd">
+   <parent>
+      <groupId>io.fabric8.dmp.samples</groupId>
+      <artifactId>dmp-sample-parent</artifactId>
+      <version>0.34-SNAPSHOT</version>
+      <relativePath>../pom.xml</relativePath>
+   </parent>
+   <modelVersion>4.0.0</modelVersion>
+   <artifactId>dmp-sample-copy-from-container</artifactId>
+   <profiles>
+      <profile>
+         <id>Copy-File</id>
+         <build>
+            <plugins>
+               <plugin>
+                  <groupId>io.fabric8</groupId>
+                  <artifactId>docker-maven-plugin</artifactId>
+                  <executions>
+                     <execution>
+                        <id>copy-from-container</id>
+                        <phase>generate-resources</phase>
+                        <goals>
+                           <goal>start</goal>
+                           <goal>copy</goal>
+                           <goal>stop</goal>
+                        </goals>
+                        <configuration>
+                           <images>
+                              <image>
+                                 <name>busybox:latest</name>
+                                 <copy>
+                                    <entries>
+                                       <entry>
+                                          <containerPath>/etc/hosts</containerPath>
+                                          <hostDirectory>/tmp/</hostDirectory>
```

We need to cover in documentation that `hostDirectory` can be relative path. In case `hostDirectory` is not an absolute path (`java.io.File#isAbsolute`) or is `null` ( is not defined), then it's considered relative path and the base directory for that relative path is the current project base directory (`org.apache.maven.project.MavenProject#getBasedir`). `null` or undefined `hostDirectory` is treated like an empty string, i.e. leads to usage of base directory of maven project.

For these samples - to make them less OS specific and to avoid spoiling filesystem of host OS with results of build - I'd recommend to use project's build directory as value of `hostDirectory`, like:

```suggestion
                                          <hostDirectory>${project.build.directory}</hostDirectory>
```

In another example we could use relative directory, like:

```suggestion
                                          <hostDirectory>target</hostDirectory>
```